ui/core/src/model/units/time.ts

```diff
@@ -74,13 +74,67 @@ export enum PersesTimeToIntlTime {
   Years = 'year',
 }
 
+// Note: This conversion will not be exactly accurate for months and years,
+// due variations in the lengths of months (i.e. 28 - 31 days) and years (i.e. leap years).
+// For precision with months and years, we would need more complex algorithms and/or external libraries.
+// However, we expect that measurements in months and years will be rare.
+const TIME_UNITS_IN_SECONDS: Record<TimeUnitKind, number> = {
+  Years: 31536000, // 365 days
+  Months: 2592000, // 30 days
+  Weeks: 604800,
+  Days: 86400,
+  Hours: 3600,
+  Minutes: 60,
+  Seconds: 1,
+  Milliseconds: 0.001,
+};
+
+const LARGEST_TO_SMALLEST_TIME_UNITS: TimeUnitKind[] = [
+  'Years',
+  'Months',
+  'Weeks',
+  'Days',
+  'Hours',
+  'Minutes',
+  'Seconds',
+  'Milliseconds',
+];
+
+// Choose the first time unit that produces a number greater than 1, starting from the biggest time unit.
```

Comment:
Optional suggestion: if you use the other style of comments, this description will show up in some text editors (e.g. vscode) when someone hovers over usage of `getValueAndKindForNaturalNumbers`, which I could see being helpful.

```ts
/**
 * Choose the first time unit that produces a number greater than 1, starting from the biggest time unit.
 */
```